app/utils/code-mirror-themes.ts

```diff
@@ -4,48 +4,231 @@ import { githubDark, githubLight } from '@uiw/codemirror-theme-github';
+      marginTop: '-0.5rem',
+      marginBottom: '0.5rem',
+
+      '&:before': {
+        borderTop: 'none',
+        marginTop: '-5px',
+        backgroundImage: 'url("/assets/images/svg-icons/expand-diff-top.svg")',
+      },
+    },
+
+    '&:last-child': {
+      borderBottom: 'none',
+      marginTop: '0.5rem',
+      marginBottom: '-0.5rem',
+
+      '&:before': {
+        borderBottom: 'none',
+        backgroundImage: 'url("/assets/images/svg-icons/expand-diff-bottom.svg")',
+      },
+    },
   },
+
   '.cm-line': {
     lineHeight: '1.5rem',
     padding: '0 1rem 0 0.625rem',
+
+    '&.cm-changedLine': {
+      backgroundColor: 'rgba(0, 255, 0, 0.15)',
+
+      '& .cm-changedText': {
+        background: 'rgba(0, 255, 0, 0.25)',
+      },
+    },
   },
+
   '.cm-deletedChunk': {
     lineHeight: '1.5rem',
     padding: '0 1rem 0 0.625rem',
+    backgroundColor: 'rgba(255, 0, 0, 0.1)',
+
+    '& del': {
+      textDecoration: 'none',
+
+      '& .cm-deletedText': {
+        background: 'rgba(255, 0, 0, 0.2)',
+      },
+    },
   },
 };
 
-export const codeCraftersLight = [EditorView.theme(Object.assign({}, BASE_STYLE), { dark: false }), githubLight];
+export const codeCraftersLight = [EditorView.theme(BASE_STYLE, { dark: false }), githubLight];
 export const codeCraftersDark = [
   EditorView.theme(
-    Object.assign({}, BASE_STYLE, {
+    {
+      '&': {
+        backgroundColor: 'rgb(15 23 42)', // bg-gray-900
+      },
+
       '.cm-gutters': {
-        backgroundColor: '#0d1117',
+        backgroundColor: 'rgb(15 23 42)', // bg-gray-900
+      },
+
+      '.cm-collapsedLines': {
+        background: 'rgb(22.8 54.2 79.4)', // bg-sky-900/40
+        color: 'rgb(56 189 248)', // text-sky-400
+        borderColor: 'rgba(255, 255, 255, 0.05)', // border-white/5
+
+        '&:before': {
+          backgroundColor: 'rgb(22.8 54.2 79.4)', // bg-sky-900/40
+          backgroundImage: 'url("/assets/images/svg-icons/expand-diff-middle-dark.svg")',
+          borderColor: 'rgba(255, 255, 255, 0.05)', // border-white/5
+        },
+
+        '&:after': {
+          backgroundColor: 'rgb(22.8 54.2 79.4)', // bg-sky-900/40
+        },
+
+        '&:hover': {
+          background: 'rgb(20.8 60.2 88.6)', // bg-sky-800/40
+          color: 'rgb(125 211 252)', // text-sky-300
+
+          '&:before': {
+            backgroundColor: 'rgb(20.8 60.2 88.6)', // bg-sky-800/40
+            color: 'rgb(125 211 252)', // text-sky-300
+          },
+
+          '&:after': {
+            backgroundColor: 'rgb(20.8 60.2 88.6)', // bg-sky-800/40
+            color: 'rgb(125 211 252)', // text-sky-300
+          },
+        },
+
+        '&:first-child': {
+          '&:before': {
+            backgroundImage: 'url("/assets/images/svg-icons/expand-diff-top-dark.svg")',
+          },
+        },
+
+        '&:last-child': {
+          '&:before': {
+            backgroundImage: 'url("/assets/images/svg-icons/expand-diff-bottom-dark.svg")',
```

Comment:
Let's place these inside `assets/images/codemirror` instead of `svg-icons`. The `svg-icons` ones are used in the app, and most of them support coloring with `currentColor`.